### PR TITLE
chore: add override routes in unkey ratelimit sdk docs

### PR DESCRIPTION
**Pull request by MichaelUnkey**
## What does this PR do?

Fixes # (issue)
Eng-1597
*If there is not an issue for this, please create one first. This is used to tracking purposes and also helps use understand why this PR exists*

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

Docs show how package works

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [Contributing Guide](./CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand areas
- [x] Ran `pnpm build`
- [x] Ran `pnpm fmt`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

## Summary by CodeRabbit

- **New Features**
	- Added comprehensive documentation for rate limiting functionalities: Set Override, Get Override, Delete Override, and List Overrides.
	- Introduced an overview document for Ratelimit overrides.
	- Enhanced the Unkey SDK documentation with new constructor options.
- **Documentation**
	- Updated existing documents for clarity and usability, including the main Ratelimit document.
	- Organized documentation structure for the `@unkey/ratelimit` library.
- **Chores**
	- Modified the `mint.json` configuration to reflect the new documentation structure.

### CODE REVIEW

File: apps/docs/libraries/ts/ratelimit/override/overview.mdx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+// handle the request here
+
+}
+```
+
+There are four main functions to interact with overrides:
+
+- [`setOverride`](http://localhost:3000/libraries/ts/ratelimit/override/set-Override) Sets an override for a ratelimit.
```
Comment (perkinsjr) [inline apps/docs/libraries/ts/ratelimit/override/overview.mdx:44]: Looks like you set this to a URL you can get rid of the localhost:3000 part.

**Review by perkinsjr — CHANGES_REQUESTED** [review]

File: apps/docs/libraries/ts/ratelimit/override/delete-override.mdx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+## Response
+
+No response, but if no error is returned the override has been deleted successfully.
+
+
+<RequestExample>
+
+```ts
+await unkey.deleteOverride({
```
Comment (perkinsjr) [inline apps/docs/libraries/ts/ratelimit/override/delete-override.mdx:32]: A request should show accurately how it works. 

Do you need all three params to delete it?

File: apps/docs/libraries/ts/ratelimit/override/get-override.mdx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+<ParamField body="async" type="boolean" required>
+Async will return a response immediately, lowering latency at the cost of accuracy.
+</ParamField>
+</Expandable>
+</ResponseField >
+
+
+<RequestExample>
+```ts
+await unkey.getOverride({
```
Comment (perkinsjr) [inline apps/docs/libraries/ts/ratelimit/override/get-override.mdx:54]: Again show a single option here. 

Then underneath another one.

File: apps/docs/libraries/ts/ratelimit/override/list-overrides.mdx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+</ResponseField>
+ </Expandable>
+
+</ResponseField>
+
+
+<RequestExample>
+  
+```ts
+await unkey.listOverrides({
```
Comment (perkinsjr) [inline apps/docs/libraries/ts/ratelimit/override/list-overrides.mdx:65]: Show just a single one. 

So it shows what is optional and how it works, you'd not send both now

File: apps/docs/libraries/ts/ratelimit/override/set-override.mdx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  </ResponseField>
+<RequestExample>
+
+```ts
+await unkey.setOverride({
+    identifier: "user_123",
+    limit: 10,
+    duration: 60000,
+    namespaceName: "email.outbound",
+    namespaceId: "rlns_1234",
```
Comment (perkinsjr) [inline apps/docs/libraries/ts/ratelimit/override/set-override.mdx:55]: Show a "create" and an update with the namespaceId for example

File: apps/docs/libraries/ts/ratelimit/override/delete-override.mdx (hunk 2 above)
Comment (MichaelUnkey) [reply apps/docs/libraries/ts/ratelimit/override/delete-override.mdx:32]: Sure dont, Just one of namespace options

File: apps/docs/libraries/ts/ratelimit/override/set-override.mdx (hunk 5 above)
Comment (MichaelUnkey) [reply apps/docs/libraries/ts/ratelimit/override/set-override.mdx]: Should we be adding some kind of header text to the example tabs?

File: apps/docs/libraries/ts/ratelimit/override/delete-override.mdx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+## Request
+<ParamField body="identifier" type="string" required>
+Identifier of your user, this can be their userId, an email, an ip or anything else. Wildcards ( * ) can be used to match multiple identifiers, More info can be found at https://www.unkey.com/docs/ratelimiting/overrides#wildcard-rules
+</ParamField>
+
+<Warning>
+Either `namespaceId` or `namespaceName` is required. Not both.
+</Warning>
+
+<ParamField body="namespaceId" type="string" required>
```
Comment (chronark) [inline apps/docs/libraries/ts/ratelimit/override/delete-override.mdx:15]: it's not required

File: apps/docs/libraries/ts/ratelimit/override/delete-override.mdx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+The id of the namespace. Either namespaceId or namespaceName must be provided
+</ParamField>
+
+<ParamField body="namespaceName" type="string" required>
```
Comment (chronark) [inline apps/docs/libraries/ts/ratelimit/override/delete-override.mdx:19]: it's not required

File: apps/docs/libraries/ts/ratelimit/override/get-override.mdx [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+## Request
+<ParamField body="identifier" type="string" required>
+Identifier of your user, this can be their userId, an email, an ip or anything else. Wildcards ( * ) can be used to match multiple identifiers, More info can be found at https://www.unkey.com/docs/ratelimiting/overrides#wildcard-rules
+</ParamField>
+
+<Warning>
+Either `namespaceId` or `namespaceName` is required. Not both.
+</Warning>
+
+<ParamField body="namespaceId" type="string" required>
```
Comment (chronark) [inline apps/docs/libraries/ts/ratelimit/override/get-override.mdx:15]: not required

File: apps/docs/libraries/ts/ratelimit/override/get-override.mdx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+The id of the namespace. Either namespaceId or namespaceName must be provided
+</ParamField>
+
+<ParamField body="namespaceName" type="string" required>
```
Comment (chronark) [inline apps/docs/libraries/ts/ratelimit/override/get-override.mdx:19]: not required

File: apps/docs/libraries/ts/ratelimit/override/get-override.mdx [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+<ParamField body="limit" type="number" required>
+How many requests may pass in a given window.
+</ParamField>
+
+<ParamField body="duration" type="number" required>
+The window duration in milliseconds.
+</ParamField>
+
+<ParamField body="async" type="boolean" required>
```
Comment (chronark) [inline apps/docs/libraries/ts/ratelimit/override/get-override.mdx:45]: not required

File: apps/docs/libraries/ts/ratelimit/override/list-overrides.mdx [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+description: "Lists all overrides"
+---
+
+## Request
+
+<Warning>
+Either `namespaceId` or `namespaceName` is required. Not both.
+</Warning>
+
+<ParamField body="namespaceId" type="string" required>
```
Comment (chronark) [inline apps/docs/libraries/ts/ratelimit/override/list-overrides.mdx:12]: not required

File: apps/docs/libraries/ts/ratelimit/override/list-overrides.mdx [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+The id of the namespace. Either namespaceId or namespaceName must be provided
+</ParamField>
+
+<ParamField body="namespaceName" type="string" required>
```
Comment (chronark) [inline apps/docs/libraries/ts/ratelimit/override/list-overrides.mdx:16]: not required

File: apps/docs/libraries/ts/ratelimit/override/get-override.mdx [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+<RequestExample>
+```ts
+await unkey.getOverride({
+  identifier:"user.example", 
+  namespaceName: "email.outbound"
+});
+
+``` 
+```ts
+await unkey.getOverride({
```
Comment (chronark) [inline apps/docs/libraries/ts/ratelimit/override/get-override.mdx:61]: make it a copy pasteable snippet
```suggestion
const override = await unkey.getOverride({
```

File: apps/docs/libraries/ts/ratelimit/override/list-overrides.mdx [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+The cursor to use for pagination
+</ResponseField>
+ </Expandable>
+
+</ResponseField>
+
+
+<RequestExample>
+```ts
+await unkey.listOverrides({
```
Comment (chronark) [inline apps/docs/libraries/ts/ratelimit/override/list-overrides.mdx:64]:
```suggestion
const overrides = await unkey.listOverrides({
```

File: apps/docs/libraries/ts/ratelimit/override/list-overrides.mdx [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+    namespaceName: "email.outbound"
+});
+```
+
+```ts
+await unkey.listOverrides({
```
Comment (chronark) [inline apps/docs/libraries/ts/ratelimit/override/list-overrides.mdx:70]:
```suggestion
const overrides = await unkey.listOverrides({
```

**Review by chronark — CHANGES_REQUESTED** [review]
nitpicks at this point

File: apps/docs/libraries/ts/ratelimit/override/delete-override.mdx [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+
+
+## Response
+
+No response, but if no error is returned the override has been deleted successfully.
+
+
+<RequestExample>
+```ts
+const override = await unkey.deleteOverride({
```
Comment (chronark) [inline apps/docs/libraries/ts/ratelimit/override/delete-override.mdx:31]: as we discussed, no variables here

File: apps/docs/libraries/ts/ratelimit/override/delete-override.mdx [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+    identifier: "user_123",
+    namespaceName: "email.outbound",
+})
+
+```
+```ts
+const override = await unkey.deleteOverride({
```
Comment (chronark) [inline apps/docs/libraries/ts/ratelimit/override/delete-override.mdx:38]: same as above

File: apps/docs/libraries/ts/ratelimit/override/overview.mdx [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+  rootKey: process.env.UNKEY_ROOT_KEY,
+})
+```
+
+## Use it
+
+```ts
+async function handler(request) {
+
+const identifier = request.getUserId() // or ip or anything else you want
```
Comment (chronark) [inline apps/docs/libraries/ts/ratelimit/override/overview.mdx:24]: indent this codeblock please, the code inside the function should be indented more

File: apps/docs/libraries/ts/ratelimit/override/overview.mdx [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+
+const override = await unkey.setOverride({
+    identifier: identifier,
+    limit: 10,
+    duration: 60000,
+    namespaceName: "email.outbound",
+    async: true
+})
+if (!override.success){
```
Comment (chronark) [inline apps/docs/libraries/ts/ratelimit/override/overview.mdx:33]: there is no `success` on the setOverride response
and why should try again later?

**Review by chronark — CHANGES_REQUESTED** [review]

File: apps/docs/libraries/ts/ratelimit/override/overview.mdx [hunk 20]
```diff
@@ -0,0 +1,48 @@
...
+      identifier: identifier,
+      limit: 10,
+      duration: 60000,
+      namespaceName: "email.outbound",
+      async: true
+  })
+  if (override.error){
+    // handle the error here
+    throw new Error(override.error.message);
+    return;
```
Comment (chronark) [inline apps/docs/libraries/ts/ratelimit/override/overview.mdx:36]: This is invalid now, after it throws the code execution stops

either throw or return, not both

**Review by chronark — CHANGES_REQUESTED** [review]

**Review by chronark — APPROVED** [review]

**Pull request merged**